### ISSUE TITLE
coq no longer reports mismatched version numbers

### ISSUE THREAD

**Issue by JasonGross** [coq/coq#12513]
#### Description of the problem

```bash
echo > foo.v
echo 'Require Import foo.' > bar.v
```
Then
```bash
~/.local64/coq/coq-8.8.0/bin/coqc -q -R . Top foo.v; ~/.local64/coq/coq-8.11.1/bin/coqc -q -R . Top bar.v
```
gives
```
File "./bar.v", line 1, characters 0-19:
Error: File /tmp/foov/foo.vo has bad magic number 8800 (expected 81100). It
is corrupted or was compiled with another version of Coq.
```
but with Coq master (831e901a8b796c3f2e7cec7f2b5d8adae4dfbea3 is the version I'm using) I get
```
File "./bar.v", line 1, characters 0-19:
Error: File /tmp/foov/foo.vo is corrupted.
```

My configuration is
```
$ ~/.local64/coq/coq-8.8.0/bin/coqc -config
LOCAL=0
COQLIB=/home/jgross/.local64/coq/coq-8.8.0/lib/coq/
DOCDIR=/home/jgross/.local64/coq/coq-8.8.0/share/doc/coq/
OCAMLFIND=/home/jgross/.opam/4.06.1+fp/bin/ocamlfind
CAMLP5O=/home/jgross/.opam/4.06.1+fp/bin/camlp5o
CAMLP5BIN=/home/jgross/.opam/4.06.1+fp/bin/
CAMLP5LIB=/home/jgross/.opam/4.06.1+fp/lib/camlp5
CAMLP5OPTIONS=-loc loc
CAMLFLAGS=-thread -rectypes -w +a-4-9-27-41-42-44-45-48-50   -safe-string
HASNATDYNLINK=true
COQ_SRC_SUBDIRS=config dev lib clib kernel library engine pretyping interp parsing proofs tactics toplevel printing intf grammar ide stm vernac plugins/btauto plugins/cc plugins/derive plugins/extraction plugins/firstorder plugins/fourier plugins/funind plugins/ltac plugins/micromega plugins/nsatz plugins/omega plugins/quote plugins/romega plugins/rtauto plugins/setoid_ring plugins/ssr plugins/ssrmatching plugins/syntax plugins/xml
$ ~/.local64/coq/coq-8.11.1/bin/coqc -config
LOCAL=0
COQLIB=/home/jgross/.local64/coq/coq-8.11.1/lib/coq/
DOCDIR=/home/jgross/.local64/coq/coq-8.11.1/share/doc/coq/
OCAMLFIND=/home/jgross/.opam/4.06.1+fp/bin/ocamlfind
CAMLFLAGS=-thread -rectypes -w +a-4-9-27-41-42-44-45-48-58-67  -bin-annot -safe-string -strict-sequence
WARN=-warn-error +a-3
HASNATDYNLINK=true
COQ_SRC_SUBDIRS=config lib clib kernel library engine pretyping interp gramlib gramlib/.pack parsing proofs tactics toplevel printing ide stm vernac plugins/btauto plugins/cc plugins/derive plugins/extraction plugins/firstorder plugins/fourier plugins/funind plugins/ltac plugins/micromega plugins/nsatz plugins/omega plugins/rtauto plugins/setoid_ring plugins/ssr plugins/ssrmatching plugins/syntax
```

This makes it significantly harder to figure out what version of Coq I compiled a given file with.  What caused this change?

#### Coq Version

master / 8.12

**Comment by Zimmi48** [comment]
This is not surprising as the vo format changed in #12092, so the old format probably looks completely foreign to the new code and the new format looks foreign to the old code. I think there is nothing we can do about the latter, but we could probably do a bit better about the former. @ppedrot WDYT?

**Comment by Zimmi48** [comment]
@JasonGross Can you confirm that the problem doesn't appear (Coq still reports a mismatched version number) when you try to load a file compiled with master in Coq 8.12+beta1 and conversely?

**Comment by ppedrot** [comment]
We could try a best effort from the newer code, upon realizing the format is not correct it could try to parse the old version number. That's a bit ugly though, and I don't know for how long we should maintain this kind of hack.

**Comment by Zimmi48** [comment]
@ppedrot Is this something that you think will be doable to fix in time for the 8.12.0 release?

**Comment by Zimmi48** [comment]
ping @ppedrot
